itest/litd_custom_channels_test.go

```diff
@@ -3304,3 +3320,220 @@ func runCustomChannelsHtlcForceClose(ctxb context.Context, t *harnessTest,
+	// assets to Fabia.
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, erinFundingAmount, assetID,
+		fn.None[int64](), lnrpc.Payment_SUCCEEDED,
+		fn.None[lnrpc.PaymentFailureReason](),
+	)
+	logBalance(t.t, nodes, assetID, "after moving assets to Fabia")
+
+	// Test case 1: We cannot keysend more assets from Erin to Fabia.
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, 1, assetID, fn.None[int64](),
+		lnrpc.Payment_FAILED, fn.Some(errNoBalance),
+	)
+
+	// Test case 2: We cannot pay an invoice from Charlie to Fabia.
+	invoiceResp := createAssetInvoice(t.t, erin, fabia, 123, assetID)
+	payInvoiceWithSatoshi(
+		t.t, charlie, invoiceResp, lnrpc.Payment_FAILED,
+		fn.Some(errNoRoute),
+	)
+
+	// Test case 3: We now create an asset buy order for a normal amount of
+	// assets. We then "fake" an invoice referencing that buy order that
+	// is for an amount that is too small to be paid with a single asset
+	// unit. This should be handled gracefully and not lead to a crash.
+	// Ideally such an invoice shouldn't be created in the first place, but
+	// we want to make sure that the system doesn't crash in this case.
+	numUnits := uint64(10)
+	buyOrderResp, err := fabiaTap.RfqClient.AddAssetBuyOrder(
+		ctxb, &rfqrpc.AddAssetBuyOrderRequest{
+			AssetSpecifier: &rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetId{
+					AssetId: assetID,
+				},
+			},
+			AssetMaxAmt: numUnits,
+			Expiry: uint64(
+				time.Now().Add(time.Hour).Unix(),
+			),
+			PeerPubKey:     erin.PubKey[:],
+			TimeoutSeconds: 10,
+		},
+	)
+	require.NoError(t.t, err)
+
+	quoteResp := buyOrderResp.Response
+	quote, ok := quoteResp.(*rfqrpc.AddAssetBuyOrderResponse_AcceptedQuote)
+	require.True(t.t, ok)
+
+	// We calculate the milli-satoshi amount one below the equivalent of a
+	// single asset unit.
+	rate, err := oraclerpc.UnmarshalFixedPoint(&oraclerpc.FixedPoint{
+		Coefficient: quote.AcceptedQuote.AskAssetRate.Coefficient,
+		Scale:       quote.AcceptedQuote.AskAssetRate.Scale,
+	})
+	require.NoError(t.t, err)
+
+	oneUnit := rfqmath.NewBigIntFixedPoint(1, 0)
+	oneUnitMilliSat := rfqmath.UnitsToMilliSatoshi(oneUnit, *rate)
+
+	t.Logf("Got quote for %v asset units per BTC", rate)
+	msatPerUnit := float64(oneUnitMilliSat) / float64(1)
+	t.Logf("Got quote for %v asset units at %3f msat/unit from peer %s "+
+		"with SCID %d", numUnits, msatPerUnit, erin.PubKeyStr,
+		quote.AcceptedQuote.Scid)
+
+	// We now manually add the invoice in order to inject the above,
+	// manually generated, quote.
+	invoiceResp2, err := fabia.AddInvoice(ctxb, &lnrpc.Invoice{
+		Memo:      "too small invoice",
+		ValueMsat: int64(oneUnitMilliSat - 1),
+		RouteHints: []*lnrpc.RouteHint{{
+			HopHints: []*lnrpc.HopHint{{
+				NodeId: erin.PubKeyStr,
+				ChanId: quote.AcceptedQuote.Scid,
+			}},
+		}},
+	})
+	require.NoError(t.t, err)
+
+	payInvoiceWithSatoshi(
+		t.t, dave, invoiceResp2, lnrpc.Payment_FAILED,
+		fn.Some(errNoRoute),
+	)
+
```

Comment:
could now do a successful payment, so that this itest is "complete"
(non-blocking: this is tested multiple times in the surrounding itests)